mmaction/core/multi_grid_utils.py

```diff
@@ -0,0 +1,252 @@
+import numpy as np
+import torch.nn as nn
+from mmcv.runner import HOOKS, Hook, LrUpdaterHook
+from mmcv.runner.hooks.lr_updater import StepLrUpdaterHook
+from torch.nn.modules.utils import _ntuple
+
+from mmaction.datasets.builder import build_dataloader
+from mmaction.datasets.pipelines.augmentations import Resize
+from mmaction.datasets.pipelines.loading import SampleFrames
+from mmaction.models.common import SubBatchBN3d
```

Comment:
```
from mmaction.datasets import build_dataloader, Resize, SampleFrames
from mmaction.models import SubBatchBN3d
```

Reply:
Resize, SampleFrames can only be imported from mmactoin.datasets.pipeliens. But yeah, will revise. BTW is that a convtion to not import from the last layer?

Reply:
Commonly import from outer layer which is no more than 2 level/